accepted/2023/LicenseInspection.md

```diff
@@ -0,0 +1,133 @@
+
+Your companies legal department has outlined a list of open source licenses that it is comfortable consuming and others that it wants to avoid using.
+
+To avoid accidentally consuming an unsupported license we need a way to view the list and ideally fail builds when we use one of these licenses.
+
+### Functional explanation
+
+#### Local scenario
+
+You have just added a new NuGet dependency into the project and want to ensure it doesn't bring in any unsupported licenses.
+
+From the command line you run the license command:
+
+```bash
+$> dotnet nuget license
+```
+
+This outputs a human readable view of the license information:
+
+```
+Project dotnet-delice
+License Expression: MIT
+├── There are 10 occurances of MIT
+├─┬ Conformance:
+│ ├── Is OSI Approved: true
+│ ├── Is FSF Free/Libre: true
+│ └── Included deprecated IDs: false
+└─┬ Packages:
+  ├── FSharp.Core
+  ├── Microsoft.NETCore.App
+  ├── Microsoft.NETCore.DotNetAppHost
+  ├── Microsoft.NETCore.DotNetHostPolicy
+  ├── Microsoft.NETCore.DotNetHostResolver
+  ├── Microsoft.NETCore.Platforms
+  ├── Microsoft.NETCore.Targets
+  ├── NETStandard.Library
+  ├── Newtonsoft.Json
+  └── System.ComponentModel.Annotations
+```
+
+Satisfied with the licenses, you continue on with the task at hand.
+
+#### Blocking builds
+
+It's time to update your build pipeline to avoid releasing anything changes that introduce unsupported licenses. A new step is added to the pipeline YAML:
+
+```yml
+- name: Validate Licenses
+  runs: |
+    LICENSES=$(dotnet nuget license --json | jq '<query for allowed licenses>')
+    if ($LICENSES) then
+        exit 1
+    fi
+```
+
+This (pseduocode version) will dump the licenses to a JSON structure that can be parsed with a tool such as [jq](https://stedolan.github.io/jq/) and inspected for invalid licenses. If any are found an error code is returned, stopping the build from completing successfully.
+
+### Technical explanation
+
+The first technical challenge for this is the inconsistent nature of which licenses are provided by NuGet packages. While the [`licenseUrl` field was deprecated](https://github.com/NuGet/Announcements/issues/32), some projects haven't adopted the new format (or older packages that predate the deprecation are in use), making it difficult to determine what the license of a project is.
+
+The next challenge is how to detect licenses from license files. The ideal approach would be to mirror GitHub's approach, which uses [Licensee](https://licensee.github.io/licensee/) [for detection](https://help.github.com/en/articles/licensing-a-repository#detecting-a-license) (but naturally a dotnet implementation). Essentially, this uses [Sørensen–Dice coefficient](https://en.wikipedia.org/wiki/S%C3%B8rensen%E2%80%93Dice_coefficient) with a threshold for what is the acceptable level of comparison between the package's license file and license template.
```

Comment:
fwiw, for anything to ever be a part of the `dotnet` commands, it needs to be source buildable: https://github.com/dotnet/source-build?tab=readme-ov-file#source-build-goals. 

We can't take a non .NET dependency easily.